### ISSUE TITLE
osqueryi extensions not showing

### ISSUE THREAD

**Issue by arirubinstein** [facebook/osquery#1018]
osqueryi with the extension socket and a custom extension fails to show the extension table when querying it interactively.  It does however show up in the osquery_extensions table.  I can reproduce this with v1.4.4 and a [demo extension](https://gist.github.com/marpaia/ddb09d4029a177850db3).

The extension appears to load and attach, but the table it implements doesn't actually appear to register in osqueryi.

I'm launching osqueryi like: `sudo osqueryi --extensions_socket /var/osquery/osquery.em`

```
osquery> select * from osquery_extensions;
+-------+------+---------+-------------+-------------------------------+-----------+
| uuid  | name | version | sdk_version | path                          | type      |
+-------+------+---------+-------------+-------------------------------+-----------+
| 0     | core | 1.4.4   | 1.4.4       | /var/osquery/osquery.em       | extension |
| 42166 | foo  |         |             | /var/osquery/osquery.em.42166 | extension |
+-------+------+---------+-------------+-------------------------------+-----------+
osquery> select * from from_extension;
Error: no such table: from_extension
```

**Comment by marpaia** [comment]
I tested this as well. The table works and is queryable in osqueryd, but not osqueryi.

**Comment by Jonty** [comment]
This is making development of new extensions a bit annoying - I'm happy to look at fixing it if someone can provide pointers as to where the problem _might_ lie?

**Comment by marpaia** [comment]
@jonty I spoke with @theopolis about this earlier, he's actively working on it. the problem seems a little more obscure than we expected, sorry for the delay.

**Comment by theopolis** [comment]
Ah, I was in the middle of refactoring the SQLite shell a few weeks ago to do just this! I got distracted when no one expressed interest in using extension-exposed tables. :/

The problem comes from how the shell code manages a sqlite in-memory database connection. When we register a new TablePlugin we need to create an SQLite module with the virtual table type and "attach" to the database instance. Right now we can't attach post-shell-repl loop. I [think](https://github.com/facebook/osquery/blob/master/osquery/devtools/shell.cpp) I remember how we were going to allow the shell to update it's DB context to reflect runtime-registered TablePlugins. Let me try first thing tomorrow morning.

Also! NB: Extensions in the shell are not autoloaded (not auto-execed as child processes) since they are [controlled by the watcher](https://github.com/facebook/osquery/blob/master/osquery/core/init.cpp#L195).

**Comment by theopolis** [comment]
Updates: there might be a quicker fix, the tables-plugins seem to fall through the shell's database instance:

```
[teddy@maverics] ~/git/osquery(branch:master) » ./build/debug_trusty/osquery/example_extension.ext --verbose
I0426 18:10:37.551964 30076 init.cpp:153] osquery initialized [version=1.4.4-146-ged69536]
I0426 18:10:37.552083 30076 extensions.cpp:163] Modules autoload contains invalid paths: /etc/osquery/modules.load
I0426 18:10:37.553191 30083 interface.cpp:249] Extension manager service starting: /home/teddy/.osquery/shell.em
I0426 18:10:37.631256 30088 events.cpp:422] Starting event publisher runloop: inotify
I0426 18:10:37.631281 30089 events.cpp:422] Starting event publisher runloop: udev
Connected to a transient in-memory database.
osquery> I0426 18:10:38.916257 30086 interface.cpp:83] Registering extension (example, 59678, version=0.0.1, sdk=1.4.4)
I0426 18:10:38.916340 30086 registry.cpp:240] Extension 59678 registered config plugin example
I0426 18:10:38.916409 30086 sqlite_util.cpp:135] DBManager contention: opening transient SQLite database
I0426 18:10:38.962002 30086 registry.cpp:240] Extension 59678 registered table plugin example
```

But are correctly added to the shell's registry:

```
osquery> select * from osquery_extensions;
+------+---------+--------------------+-------------+-----------------------------------+-----------+
| uuid | name    | version            | sdk_version | path                              | type      |
+------+---------+--------------------+-------------+-----------------------------------+-----------+
| 0    | core    | 1.4.4-146-ged69536 | 1.4.4       | /home/teddy/.osquery/shell.em     | extension |
| 961  | example | 0.0.1              | 1.4.4       | /home/teddy/.osquery/shell.em.961 | extension |
+------+---------+--------------------+-------------+-----------------------------------+-----------+

osquery> select * from osquery_registry;
+------------------+----------------------+------------+----------+--------+
| registry         | name                 | owner_uuid | internal | active |
+------------------+----------------------+------------+----------+--------+
| config           | filesystem           | 0          | 0        | 1      |
| config           | http                 | 0          | 0        | 1      |
| config           | update               | 0          | 0        | 1      |
| config           | example              | 6056       | 0        | 1      |
| config_parser    | yara                 | 0          | 0        | 1      |
| enrollment       | get_key              | 0          | 0        | 1      |
| event_publisher  | inotify              | 0          | 0        | 1      |
| event_publisher  | udev                 | 0          | 0        | 1      |
| event_subscriber | file_changes         | 0          | 0        | 1      |
| event_subscriber | hardware_events      | 0          | 0        | 1      |
| event_subscriber | passwd_changes       | 0          | 0        | 1      |
| event_subscriber | yara_events          | 0          | 0        | 1      |
| logger           | filesystem           | 0          | 0        | 1      |
| sql              | sql                  | 0          | 1        | 1      |
| table            | acpi_tables          | 0          | 0        | 1      |
| table            | apt_sources          | 0          | 0        | 1      |
| table            | arp_cache            | 0          | 0        | 1      |
| table            | block_devices        | 0          | 0        | 1      |
| table            | chrome_extensions    | 0          | 0        | 1      |
| table            | cpuid                | 0          | 0        | 1      |
| table            | crontab              | 0          | 0        | 1      |
| table            | deb_packages         | 0          | 0        | 1      |
| table            | disk_encryption      | 0          | 0        | 1      |
| table            | etc_hosts            | 0          | 0        | 1      |
| table            | etc_protocols        | 0          | 0        | 1      |
| table            | etc_services         | 0          | 0        | 1      |
| table            | file                 | 0          | 1        | 1      |
| table            | file_changes         | 0          | 0        | 1      |
| table            | firefox_addons       | 0          | 0        | 1      |
| table            | groups               | 0          | 0        | 1      |
| table            | hardware_events      | 0          | 0        | 1      |
| table            | hash                 | 0          | 1        | 1      |
| table            | interface_addresses  | 0          | 0        | 1      |
| table            | interface_details    | 0          | 0        | 1      |
| table            | kernel_info          | 0          | 0        | 1      |
| table            | kernel_integrity     | 0          | 0        | 1      |
| table            | kernel_modules       | 0          | 0        | 1      |
| table            | last                 | 0          | 0        | 1      |
| table            | listening_ports      | 0          | 0        | 1      |
| table            | logged_in_users      | 0          | 0        | 1      |
| table            | memory_map           | 0          | 0        | 1      |
| table            | mounts               | 0          | 0        | 1      |
| table            | opera_extensions     | 0          | 0        | 1      |
| table            | os_version           | 0          | 0        | 1      |
| table            | osquery_extensions   | 0          | 1        | 1      |
| table            | osquery_flags        | 0          | 1        | 1      |
| table            | osquery_info         | 0          | 1        | 1      |
| table            | osquery_registry     | 0          | 1        | 1      |
| table            | osquery_schedule     | 0          | 1        | 1      |
| table            | passwd_changes       | 0          | 0        | 1      |
| table            | pci_devices          | 0          | 0        | 1      |
| table            | process_envs         | 0          | 0        | 1      |
| table            | process_memory_map   | 0          | 0        | 1      |
| table            | process_open_files   | 0          | 0        | 1      |
| table            | process_open_sockets | 0          | 0        | 1      |
| table            | processes            | 0          | 0        | 1      |
| table            | routes               | 0          | 0        | 1      |
| table            | shared_memory        | 0          | 0        | 1      |
| table            | shell_history        | 0          | 0        | 1      |
| table            | smbios_tables        | 0          | 0        | 1      |
| table            | suid_bin             | 0          | 0        | 1      |
| table            | system_controls      | 0          | 0        | 1      |
| table            | time                 | 0          | 1        | 1      |
| table            | usb_devices          | 0          | 0        | 1      |
| table            | users                | 0          | 0        | 1      |
| table            | yara                 | 0          | 0        | 1      |
| table            | yara_events          | 0          | 0        | 1      |
| table            | example              | 6056       | 0        | 1      |
+------------------+----------------------+------------+----------+--------+
```

In this case I am running/debugging with the example (C++) extension built inline with osqueryi:

```
[teddy@maverics] ~/git/osquery(branch:master) » ./build/debug_trusty/osquery/example_extension.ext --verbose --socket ~/.osquery/shell.em
I0426 18:10:38.915285 30090 init.cpp:155] osquery extension initialized [sdk=1.4.4]
I0426 18:10:38.964251 30090 extensions.cpp:329] Extension (example, 59678, 0.0.1, 1.4.4) registered
I0426 18:10:38.964427 30096 interface.cpp:204] Extension service starting: /home/teddy/.osquery/shell.em.59678
```

**Comment by theopolis** [comment]
Should be fixed by: #1051
